core/src/test/java/org/apache/iceberg/rest/TestHTTPClient.java

```diff
@@ -121,6 +128,92 @@ public void testHeadFailure() throws JsonProcessingException {
     testHttpMethodOnFailure(HttpMethod.HEAD);
   }
 
+  /** Tests that requests go via the proxy server in case the client is set up with one */
+  @Test
+  public void testHttpClientProxyServerInteraction() throws IOException {
+    int proxyPort = 1070;
+    String proxyHostName = "localhost";
+    try (ClientAndServer proxyServer = startClientAndServer(proxyPort);
+        RESTClient clientWithProxy =
+            HTTPClient.builder(ImmutableMap.of())
+                .uri(URI)
+                .withProxy(proxyHostName, proxyPort)
+                .build()) {
+      //  Set up the servers to match against a provided request
+      String path = "v1/config";
+
+      HttpRequest mockRequest =
+          request("/" + path).withMethod(HttpMethod.HEAD.name().toUpperCase(Locale.ROOT));
+
+      HttpResponse mockResponse = response().withStatusCode(200);
+
+      mockServer.when(mockRequest).respond(mockResponse);
+      proxyServer.when(mockRequest).respond(mockResponse);
+
+      restClient.head(path, ImmutableMap.of(), (onError) -> {});
+      mockServer.verify(mockRequest, VerificationTimes.exactly(1));
+      proxyServer.verify(mockRequest, VerificationTimes.never());
```

Comment:
I see the intent but I don't think we need this validation since other tests will validate this. The reason is, I think this verification becomes the equivalent of "restClient should be hitting mockServer directly, and there shouldn't be some other server in between" and the other tests will validate that just by leveraging the response that mockServer returns.

Reply:
Makes sense